[Communication]: Update android release notes for communication chat and common beta7 r… (#2534)
Azure/azure-sdk: releases/2021-03/android.md

### Gradle

#### Java

```gradle
allprojects {
    repositories {
        ...
        maven {
            url 'https://trouterpublicpackages.z13.web.core.windows.net'
        }
    }

dependencies {
    ...

>     implementation 'com.azure.android:azure-communication-chat:1.0.0-beta.7'

Reviewer: should we add common package here and below?

### Gradle

#### Java

```gradle
allprojects {
    repositories {
        ...
        maven {
            url 'https://trouterpublicpackages.z13.web.core.windows.net'
        }
    }

dependencies {
    ...
    implementation 'com.azure.android:azure-communication-chat:1.0.0-beta.7'
    implementation 'com.azure.android:azure-communication-common:1.0.0-beta.7'
}
```

#### Kotlin

```gradle
allprojects {
    repositories {
        ...
        maven {
            url 'https://trouterpublicpackages.z13.web.core.windows.net'
        }
    }
dependencies {
    ...
    implementation("com.azure.android:azure-communication-chat:1.0.0-beta.7")

>     implementation("com.azure.android:azure-communication-common:1.0.0-beta.7")

Reviewer 2: Does any of your customer consume the `common` package as a standalone dependency? If so, these examples are valid. If not, it's prolly not event worth mentioning `common` as it'll be automatically managed for them? Just a thought.

You might also just want to separate the scenario and make it clear to your customers that if they want `chat`, then `common` is automatically included, etc.. .

>     implementation("com.azure.android:azure-communication-common:1.0.0-beta.7")

Reviewer 3: This section in the release notes serves to show users how they can import the latest version of the packages we release, we even include Azure Core in these. That said, I see what you mean by possibly making it confusing to users when they read this without said context though. What do you think about that @czubair? Maybe we could add a legend saying `common` is used by `chat` and `core` is used by everything? I feel like this would just make things too complicated. I mean, normally I would expect people's first point of contact with our SDKs to be the READMEs, but I know that won't always be the case.

>     implementation("com.azure.android:azure-communication-common:1.0.0-beta.7")

Reviewer 3: We have added `common` to release notes in the past, I don't see any harm in keeping it like we do with Core and Identity in other languages. The way I see it, it's open source after all and people can build whatever they want on top of it.

>     implementation("com.azure.android:azure-communication-common:1.0.0-beta.7")

Reviewer 4: In JS, we don't have core in our release notes. But we do have communication-common in the release notes. This is an example of the Feb release - https://github.com/Azure/azure-sdk/pull/2333/files

>     implementation("com.azure.android:azure-communication-common:1.0.0-beta.7")

Reviewer 4: Also just thought of sharing it here - https://github.com/Azure/azure-sdk/blob/08f279b39a4c19822b1509e6b10fc3ce256607b5/docs/policies/releasenotes.md#whats-a-developer-impacting-change

## Release highlights

### Azure Communication Services Common

#### 1.0.0-beta.7 ([Changelog](https://github.com/Azure/azure-sdk-for-android/blob/master/sdk/communication/azure-communication-common/CHANGELOG.md#100-beta7-2021-03-09))

##### Breaking Changes

> - Credential `getToken` returns the newly added `CommunicationAccessToken` object instead of `AccessToken`.
> - Renamed 'getRefreshProactively' to 'isRefreshProactively' in 'CommunicationTokenRefreshOptions'
> - Removed constructor 'MicrosoftTeamsUserIdentifier(String userId, boolean isAnonymous, CommunicationCloudEnvironment cloudEnvironment)' in 'MicrosoftTeamsUserIdentifier'
> - A few classes are made final 'CommunicationTokenCredential', 'CommunicationTokenRefreshOptions', 'CommunicationUserIdentifier', 'MicrosoftTeamsUserIdentifier', 'PhoneNumberIdentifier', 'UnknownIdentifier', 'CommunicationAccessToken'

Reviewer 3: ```suggestion
- Credential `getToken` now returns the newly added `CommunicationAccessToken` object instead of `AccessToken`.
- Renamed `getRefreshProactively` to `isRefreshProactively` in 'CommunicationTokenRefreshOptions'.
- Removed constructor `MicrosoftTeamsUserIdentifier(String userId, boolean isAnonymous, CommunicationCloudEnvironment cloudEnvironment)` in `MicrosoftTeamsUserIdentifier`.
- Made the following classes final:
    - `CommunicationAccessToken`
    - `CommunicationTokenCredential`
    - `CommunicationTokenRefreshOptions`
    - `CommunicationUserIdentifier`
    - `MicrosoftTeamsUserIdentifier`
    - `PhoneNumberIdentifier`
    - `UnknownIdentifier`
```

##### New Features

> - Introduce new class `CommunicationAccessToken`.

Reviewer 3: ```suggestion
- Introduced a new class: `CommunicationAccessToken`.
```

### Azure Communication Services Chat

#### 1.0.0-beta.7 ([Changelog](https://github.com/Azure/azure-sdk-for-android/blob/master/sdk/communication/azure-communication-chat/CHANGELOG.md#100-beta7-2021-03-09))

> ##### Added

Reviewer 3: ```suggestion
##### New Features
```

- Support real time notifications with new methods in ChatClient/ChatAsyncClient: 
    - startRealtimeNotifications
    - stopRealtimeNotifications
    - on(chatEventId, listenerId, listener) 
    - off(chatEventId, listenerId) 
- Add a sample chat app under folder samples for testing and playing around chat functionality purpose.

##### Breaking Changes

> - Change remove participant API to /chat/threads/{chatThreadId}/participants/:remove
> - user id in following classes changed from type CommunicationUserIdentifier to type CommunicationIdentifierModel.
> - property initiator in ChatMessageContent renamed to initiatorCommunicationIdentifier.
> - property senderId in ChatMessage and ChatMessageReadReceipt renamed to senderCommunicationIdentifier.
> - property identifier in ChatParticipant renamed to communicationIdentifier.
> - property createdBy in ChatThread renamed to createdByCommunicationIdentifier.
> - repeatability-Request-ID in header renamed to repeatability-Request-Id.

Reviewer 3: We don't need the first and last points in the release notes as they not part of the public API but internal implementation.

```suggestion
- Changed `userId` from the type `CommunicationUserIdentifier` to `CommunicationIdentifierModel` in the following classes:
    - `ChatMessage`
    - `ChatMessageContent`
    - `ChatMessageReadReceipt`
    - `ChatParticipant`
    - `ChatThread`
- Renamed property `initiator` to `initiatorCommunicationIdentifier` in `ChatMessageContent`.
- Renamed `senderId` to `senderCommunicationIdentifier` in `ChatMessage` and `ChatMessageReadReceipt`.
- Renamed `identifier` to `communicationIdentifier` in `ChatParticipant`.
- Renamed `createdBy` to `createdByCommunicationIdentifier` in `ChatThread`.
```
